src/providers/mssql/qgsmssqlproviderconnection.cpp

```diff
@@ -0,0 +1,482 @@
+             .arg( QgsMssqlProvider::quotedValue( schema ) );
+  }
+  else
+  {
+    query += QStringLiteral( R"raw(
+                             sys.schemas.name, sys.objects.name, sys.columns.name, null, 'GEOMETRY', CASE WHEN sys.objects.type = 'V' THEN 1 ELSE 0 END
+                              FROM sys.columns
+                                JOIN sys.types
+                                  ON sys.columns.system_type_id = sys.types.system_type_id AND sys.columns.user_type_id = sys.types.user_type_id
+                                JOIN sys.objects
+                                  ON sys.objects.object_id = sys.columns.object_id
+                                JOIN sys.schemas
+                                  ON sys.objects.schema_id = sys.schemas.schema_id
+                                WHERE
+                                  sys.schemas.name = N%1
+                                  AND (sys.types.name = 'geometry' OR sys.types.name = 'geography')
+                                  AND (sys.objects.type = 'U' OR sys.objects.type = 'V')
+                             )raw" )
+             .arg( QgsMssqlProvider::quotedValue( schema ) );
+  }
+
+  if ( allowGeometrylessTables )
+  {
+    query += QStringLiteral( R"raw(
+                             UNION ALL SELECT sys.schemas.name, sys.objects.name, null, null, 'NONE',
+                               CASE WHEN sys.objects.type = 'V' THEN 1 ELSE 0 END
+                             FROM sys.objects
+                               JOIN sys.schemas
+                                  ON sys.objects.schema_id = sys.schemas.schema_id
+                             WHERE
+                               sys.schemas.name = N%1
+                               AND NOT EXISTS
+                                (SELECT *
+                                  FROM sys.columns sc1 JOIN sys.types ON sc1.system_type_id = sys.types.system_type_id
+                                  WHERE (sys.types.name = 'geometry' OR sys.types.name = 'geography')
+                                    AND sys.objects.object_id = sc1.object_id )
+                               AND (sys.objects.type = 'U' OR sys.objects.type = 'V')
+                             )raw" )
+             .arg( QgsMssqlProvider::quotedValue( schema ) );
+  }
+
+  const QList<QVariantList> results { executeSqlPrivate( query, false ) };
+  for ( const auto &row : results )
+  {
+    Q_ASSERT( row.count( ) == 6 );
+    QgsMssqlProviderConnection::TableProperty table;
+    table.setSchema( row[0].toString() );
+    table.setTableName( row[1].toString() );
+    table.setGeometryColumn( row[2].toString() );
+    //const QVariant srid { row[3] };
+    //const QVariant type { row[4] }; // GEOMETRY|GEOGRAPHY
+    if ( row[5].toBool() )
+      table.setFlag( QgsMssqlProviderConnection::TableFlag::View );
+
+    int geomColCount { 0 };
+
+    if ( ! table.geometryColumn().isEmpty() )
```

Comment:
How do you feel about adding a flag "GeometryColumnDetailsOptional" ="in cases where resolving the full geometry details of a table are expensive, setting this flag will skip fetching these details"

Reply:
That's a good idea, there are cases where you just need the table names and you don't care about other details. But I would leave this to a separate PR because it impacts all other providers.